replication-api/src/main/java/org/codice/ditto/replication/api/persistence/ReplicatorConfigManager.java

```diff
@@ -39,13 +50,17 @@
    * one. Else, saves the {@link ReplicatorConfig}.
    *
    * @param replicationConfig The configuration to save or update
+   * @throws ReplicationPersistenceException if an error occurs while trying to save the config
+   * @throws IllegalArgumentException if the ReplicatorConfig implementation is not one that can be
+   *     saved
    */
   void saveConfig(ReplicatorConfig replicationConfig);
```

Comment:
if it has the same name as another config that should be recognized and the save request should be rejected at the boundary. I don't think we want to try and handle that everywhere. If it has the same id as another config then this will overwrite that config.

Reply:
Same reply as before, the boundary can only validate for one admin at a time. If 2 admins are updating configs then boundary checking is no longer valid and the database can be corrupted with duplicate names.